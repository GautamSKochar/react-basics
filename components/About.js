import React from 'react';
import ReactDOM from 'react-dom';

var panel = {
    marginBottom:'20px',
    fontSize:'12px',
    border:'1px solid transparent',
    borderColor: '#d6e9c6',
    borderRadius: '4px',
    boxShadow: '0 1px 1px rgba(0,0,0,.05)'
}
var panelHeading ={
    'padding': '10px 15px',
    'borderBottom': '1px solid transparent',
    'borderTopLeftRadius': '3px',
    'borderTopRightRadius': '3px',
    'color': '#3c763d',
    'backgroundColor': '#dff0d8',
    'borderColor': '#d6e9c6'
}

var panelBody ={
    padding: '15px'
}

class About extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loading: true,
            error: null,
            data: null
        };
    }

    componentDidMount() {
        this.props.promise.then(
            value => this.setState({ loading: false, data: value }),
            error => this.setState({ loading: false, error: error }));
    }

    render() {
        if (this.state.loading) {
            return <div style={{position:'fixed', top:'35%', left: '40%' }}><img src="../assets/loading.gif" alt="loading..."></img></div>;
        }
        else if (this.state.error !== null) {
            return <span>Error: {this.state.error.message}</span>;
        }
        else {
            var repos = this.state.data.items;
            var repoList = repos.map(function (repo,i) {
                return (

                    
                    <div key={i} className="panel panel-success" style={panel}>
                        <div className="panel-heading" style={panelHeading}>
                            <h4 className="panel-title"><a href={repo.html_url}>{repo.name}</a>&nbsp;&nbsp;&nbsp;({repo.stargazers_count} stars)</h4>
                        </div>
                        <div className="panel-body">
                            {repo.description}
                        </div>
                    </div>
                );
            });
            return (
                <main>
                    <h3>About Us...Random Dummy Data</h3>
                    <article>{repoList}</article>
                </main>
            );
        }


    }
}
export default About;