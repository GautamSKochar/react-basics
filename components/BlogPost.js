import React from 'react';
import post from '../blogpost.json';

const BlogPost = () => (
    <div>
        {post.posts.map((post,i) => <div key={i}>{post.title}</div>)}
    </div>
)

export default BlogPost