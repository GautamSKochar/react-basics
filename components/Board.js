import React from 'react';
import Featured from './Features';
import { connect } from 'react-redux';
import { addComment, editComment, deleteComment } from '../actions/commentActions';

class Board extends React.Component {

    addNew(text) {
        this.props.addComment(text);
    }

    removeComment(i) {
        this.props.deleteComment(i);
    }

    updateComment(newText, i) {
        this.props.editComment(i, newText);
    }

    eachComment(txt, i) {
        return (
            <Featured key={i} index={i}
                updateCommentText={this.updateComment.bind(this)}
                deleteFromBoard={this.removeComment.bind(this)}>{txt}</Featured>
        );
    }

    render() {
        return (
            <div className="container" style={{ float: 'left', display: 'inline-block', whiteSpace: 'nowrap' }}>
                <h3>Board of comments</h3>
                <hr />
                <button onClick={() => this.addNew("newww text")} className="btn btn-info">Add New</button>
                <hr />
                {this.props.comments.map(this.eachComment.bind(this))}
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        comments: state.comments.comments
    };
}

const mapDispatchToPropss = (dispatch) => {
    return {
        addComment: (newText) => {
            dispatch(addComment(newText))
        },
        editComment: (id, newText) => {
            dispatch(editComment(id, newText))
        },
        deleteComment: (id) => {
            dispatch(deleteComment(id))
        }
    };
}

export default connect(mapStateToProps, mapDispatchToPropss)(Board);