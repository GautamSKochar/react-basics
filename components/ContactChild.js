import React from 'react';

class ContactChild extends React.Component {
    constructor() {
        super();
        this.state = {
            // pageMounted: true,
            opacity: 1.0
        };
        console.log('Constructor');
    }

    componentWillMount() {
        console.log('Component will mount');
    };
    componentDidMount() {
        console.log('Component did mount');
        this.timer = setInterval(function () {
            var opacity = this.state.opacity;
            opacity = parseFloat(opacity) - .05;
            if (opacity < 0.1) {
                opacity = 1.0;
            }
            this.setState({
                opacity: opacity
            });
        }.bind(this), 100);
    };
    componentWillReceiveProps(nextProps) {
        console.log('Component will receive props', nextProps);
    };
    shouldComponentUpdate(nextProps, nextState) {
        console.log('Should Component update', nextProps, nextState);
        return true;
    }
    componentWillUpdate(nextProps, nextState) {
        console.log('Component will update', nextProps, nextState);
    }
    componentDidUpdate(prevProps, prevState) {
        console.log('Component Did Update', prevProps, prevState);
    }

    componentWillUnmount() {
        console.log("Component Will Unmount");
    }

    render() {
        
        return (
            
            <div className="well" style={{marginLeft: '20px', }}>
                <h4>Child Component</h4><br/>
                <h5>Component Lifecycle Examples</h5>
                <section>Some dummy text...................</section>
                <div style={{ opacity: this.state.opacity }}>
                    <h5>Hello React...! Check Console for Component life cycle methods.</h5>
                </div>                
            </div>
        )
    }
}

export default ContactChild;