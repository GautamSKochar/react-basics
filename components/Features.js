import React from 'react';

var noteStyle = {
    'textDecoration': 'none',
    'color': '#999',
    'display': 'block',
    'height': '120px',
    'width': '260px',
    'padding': '1em',
    'MozBoxShadow': '5px 5px 7px #212121',
    'WebkitBoxShadow': '5px 5px 7px rgba(33, 33, 33, 0.7)',
    'boxShadow': '5px 5px 7px rgba(33, 33, 33, 0.7)',
    'MozTransition': '-moz-transform 0.15s linear',
    'OTransition': '-o-transform 0.15s linear',
    'WebkitTransition': '-webkit-transform 0.15s linear',
    'backgroundColor': 'aliceblue',
    'overflow':'auto'
};

var divContainer = {
    display: 'inline',
    float: 'left',
    maxWidth: '350px'
};

class Featured extends React.Component {
    constructor(props) {
        super();
        this.state = { 
            editing: false
        }
    }
    onEdit() {
        this.setState({
            editing: true
        });
        console.log('changing state');
    }
    onRemove() {
        this.props.deleteFromBoard(this.props.index);
    }
    onCancel() {
        this.setState({ editing: false });
    }
    onSave(e) {
        this.props.updateCommentText(this.refs.nexText.value, this.props.index);
        this.setState({ 
            editing: false
        });
    }

    normalRender(){
        return (
            <div className="container" style={divContainer}>
                <div style={noteStyle} index={this.props.index}>{this.props.children}</div>
                <div className="btn-group" style={{ paddingTop: '5px', paddingLeft: '25px' }}>
                    <button id="editBtn" onClick={this.onEdit.bind(this)} className="btn btn-primary">Edit</button>
                    <button onClick={this.onRemove.bind(this)} className="btn btn-danger">Delete</button>
                </div>
                <br/><br/><br/>
            </div>
        );
    }
    renderForm(){
        return (
            <div className="container" style={divContainer}>
                <textarea ref="nexText" rows="4" defaultValue={this.props.children}></textarea>
                <br/>
                <div className="btn-group" style={{ paddingTop: '5px', paddingLeft: '25px' }}>
                    <button onClick={this.onSave.bind(this)} className="btn btn-success">Save</button>
                    <button onClick={this.onCancel.bind(this)} className="btn btn-defalut">Cancel</button>
                </div>
            </div>
        );
    }

    render() {
        if (this.state.editing) {
            return this.renderForm.bind(this)();
        } else {
           return this.normalRender.bind(this)();
        }

    }
}

export default Featured;