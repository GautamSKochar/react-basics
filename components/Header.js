import React from "react";
import {Link} from 'react-router-dom';

export const Header = (props) => {
    return (
        <div className="page-header" style={{backgroundColor:'lightgrey'}}>
            <h1 style={{textAlign:'center'}}>React POC Demo</h1>
            <nav className="navbar navbar-default navbar-fixed-top">
                {/* <div className="container"> */}
                {/* <div className="navbar-header"> */}
                            <ul className="header">
                                <li><Link to="/">Home</Link></li>
                                <li><Link to="/features">Features</Link></li>
                                <li><Link to="/about">About</Link></li>
                                <li><Link to="/contact">Contact</Link></li>
                                <li><Link to="/user/4">User</Link></li>                                
                            </ul>
                        {/* </div> */}
                {/* </div> */}
            </nav>
        </div>


    );

}