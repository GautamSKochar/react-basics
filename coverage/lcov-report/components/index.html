<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">53.1% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>60/113</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>30/60</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.05% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>60/111</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="About.js"><a href="About.js.html">About.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="16" class="abs medium">8/16</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="16" class="abs medium">8/16</td>
	</tr>

<tr>
	<td class="file high" data-value="BlogPost.js"><a href="BlogPost.js.html">BlogPost.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file low" data-value="Board.js"><a href="Board.js.html">Board.js</a></td>
	<td data-value="28.57" class="pic low"><div class="chart"><div class="cover-fill" style="width: 28%;"></div><div class="cover-empty" style="width:72%;"></div></div></td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	</tr>

<tr>
	<td class="file high" data-value="Contact.js"><a href="Contact.js.html">Contact.js</a></td>
	<td data-value="85.71" class="pic high"><div class="chart"><div class="cover-fill" style="width: 85%;"></div><div class="cover-empty" style="width:15%;"></div></div></td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

<tr>
	<td class="file low" data-value="ContactChild.js"><a href="ContactChild.js.html">ContactChild.js</a></td>
	<td data-value="38.89" class="pic low"><div class="chart"><div class="cover-fill" style="width: 38%;"></div><div class="cover-empty" style="width:62%;"></div></div></td>
	<td data-value="38.89" class="pct low">38.89%</td>
	<td data-value="18" class="abs low">7/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	<td data-value="38.89" class="pct low">38.89%</td>
	<td data-value="18" class="abs low">7/18</td>
	</tr>

<tr>
	<td class="file medium" data-value="Features.js"><a href="Features.js.html">Features.js</a></td>
	<td data-value="73.33" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 73%;"></div><div class="cover-empty" style="width:27%;"></div></div></td>
	<td data-value="73.33" class="pct medium">73.33%</td>
	<td data-value="15" class="abs medium">11/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="73.33" class="pct medium">73.33%</td>
	<td data-value="15" class="abs medium">11/15</td>
	</tr>

<tr>
	<td class="file high" data-value="Header.js"><a href="Header.js.html">Header.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file low" data-value="Home.js"><a href="Home.js.html">Home.js</a></td>
	<td data-value="38.89" class="pic low"><div class="chart"><div class="cover-fill" style="width: 38%;"></div><div class="cover-empty" style="width:62%;"></div></div></td>
	<td data-value="38.89" class="pct low">38.89%</td>
	<td data-value="18" class="abs low">7/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="38.89" class="pct low">38.89%</td>
	<td data-value="18" class="abs low">7/18</td>
	</tr>

<tr>
	<td class="file high" data-value="Layout.jsx"><a href="Layout.jsx.html">Layout.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="TodoList.js"><a href="TodoList.js.html">TodoList.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file low" data-value="User.js"><a href="User.js.html">User.js</a></td>
	<td data-value="46.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 46%;"></div><div class="cover-empty" style="width:54%;"></div></div></td>
	<td data-value="46.67" class="pct low">46.67%</td>
	<td data-value="15" class="abs low">7/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="53.85" class="pct medium">53.85%</td>
	<td data-value="13" class="abs medium">7/13</td>
	</tr>

<tr>
	<td class="file high" data-value="sum.js"><a href="sum.js.html">sum.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 11 2017 16:10:38 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
