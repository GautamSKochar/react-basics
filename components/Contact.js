import React from 'react';
import ContactChild from './ContactChild';

class Contact extends React.Component {
    constructor() {
        super();
        this.state = {
            pageMounted: true
        };
    }

    onChangePageMounted() {
        this.setState({
            pageMounted: !this.state.pageMounted
        });
    };
    render() {
        let pageCmp = "";
        if (this.state.pageMounted) {
            pageCmp = <div>
                <ContactChild/>
            </div>;
        }
        return (
            <div>
                <h3>Contact Us</h3><br/>
                {pageCmp}<br/>
                <div>
                    <button onClick={this.onChangePageMounted.bind(this)} className="btn btn-primary">(Un)Mount Child Component</button>
                </div>
            </div>
        )
    }
}

export default Contact;