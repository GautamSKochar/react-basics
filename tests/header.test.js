import React from 'react';
// import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import { Header } from '../components/Header';
// setup file
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}

configure({ adapter: new Adapter() });

test('Header Component Snapshot Test', () => {
    const component = shallow(<Header />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});    