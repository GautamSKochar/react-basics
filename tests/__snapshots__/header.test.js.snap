// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot Test 1`] = `
<div
  className="page-header"
  style={
    Object {
      "backgroundColor": "lightgrey",
    }
  }
>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    ReactJS POC
  </h1>
  <nav
    className="navbar navbar-default navbar-fixed-top"
  >
    <ul
      className="header"
    >
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/features"
        >
          Features
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/about"
        >
          About
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/contact"
        >
          Contact
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/user/4"
        >
          User
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/products"
        >
          Products
        </Link>
      </li>
    </ul>
  </nav>
</div>
`;
